src/nupic/algorithms/KNNClassifier.py

```diff
@@ -845,31 +869,35 @@ def getPartitionId(self, i):
 
   def getPartitionIdPerPattern(self):
     """
-    Returns a list of numPatterns elements where the i'th position contains
-    the integer partition Id associated with pattern i. If pattern i had no
-    partition Id, it's value will be numpy.inf
+    Gets all the partition ids.
+
+    TODO: This method doesn't make sense to me. --Matt
+
+    :returns: a list of numPatterns elements where the i'th position contains
+        the integer partition Id associated with pattern i. If pattern i had no
+        partition Id, it's value will be numpy.inf
```

Comment:
The only place I see this method used is within a unit test:

https://github.com/numenta/nupic/blob/3680b5589fc1a84c3eb1fd6f9b8e7de1104cc31c/tests/unit/nupic/algorithms/knn_classifier_test.py#L355-L358

I recommend we remove the function, unless @subutai has a reason to leave it.